### DISCUSSION TITLE
ERROR WHILE RUNNING

### DISCUSSION THREAD

**Discussion by sonukumar005** [oppia/oppia-android#5581] [Q&A (Installation)] [answered]
### Operating System

WINDOWS

### Android Studio Version

Android Studio Bumblebee | 2021.1.1 Patch 3

### RAM size

8 GB

### Free disk space

198 GB

### Which step are you stuck on?

when ever i try to run apk in my device by ubs  it show this error

![Screenshot 2024-11-18 201838](https://github.com/user-attachments/assets/26bbe44d-68e4-4f11-a25f-4dd197c96287)
![Screenshot 2024-11-18 201800](https://github.com/user-attachments/assets/452a56cc-f3b2-47b3-ba7c-d12ad7651656)
![Screenshot 2024-11-18 201901](https://github.com/user-attachments/assets/7fe8b3a3-4404-4b1a-9857-5195ad50f7f0)

### Error Log

```bash
Error: Execution failed for task ':app:packageDebug'.

Cause: A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
Inner Cause: com.android.ide.common.signing.KeytoolException: Failed to read key AndroidDebugKey from store "C:\Users\Sonu.android\debug.keystore": Integrity check failed: java.security.NoSuchAlgorithmException: Algorithm HmacPBESHA256 not available
Caused by:

java.security.NoSuchAlgorithmException: Algorithm HmacPBESHA256 not available



### Approaches already used to resolve the issue

_No response_

### Additional information

_No response_

**Comment by KaranJoshi1208** [comment, marked as answer]
Delete the `debug.keystore ` file in location `C:\User\Sonu\.android` , and rebuild the project .

### I don't know why but it works.

If anyone knows the exact reason behind this error, please share your insights with us.

**Reply by sonukumar005** [reply]
it worked

**Comment by KaranJoshi1208** [comment]
![image](https://github.com/user-attachments/assets/bab12d87-3af0-4d21-83ee-c958b4f7cf5f)


#### Build is successful, but does anyone know what are those errors ?

**Comment by KaranJoshi1208** [comment]
![image](https://github.com/user-attachments/assets/a81cac29-e63b-402e-bcc6-04467342ced9)

### Gradle sync failed ??

**Comment by adhiamboperes** [comment]
The error indicates that Android's build process failed because the signing key used for debug builds (debug.keystore) couldn't be read. Specifically, the issue is due to the algorithm `HmacPBESHA256`, which is required to access the **debug.keystore**, being unavailable in your Java environment.

This is likely due to incompatible Java versions. For example, you might have run an Android Studio build with a newer version of Java, before using the Oppia-supported version.

**Reply by KaranJoshi1208** [reply]
Ah , I see. 
I made it work by changing the Android Studio verion from BumbleBee to ElectricEle  .

Thank you @adhiamboperes for the explanation 👍

**Comment by Varshith-Yadav** [comment]
Hi everyone,

I am facing an issue where the AAPT process unexpectedly exits while building my Android project. Below are the details of my setup and the error message:
`AAPT2 process unexpectedly exit. Error output:
aapt2.exe W 02-14 00:27:25  5240 26900 LoadedArsc.cpp:682] Unknown chunk type '200'.
`

Has anyone encountered this issue before? Any suggestions on how to resolve it?

Thanks in advance for your help!

**Reply by TanishMoral11** [reply]
Hi @Varshith-Yadav,

Please refer to the following discussions :

- [Discussion 5672](https://github.com/oppia/oppia-android/discussions/5672)
- [Discussion 5632](https://github.com/oppia/oppia-android/discussions/5632)

Please feel free to ask if you have any doubts.

**Comment by Varshith-Yadav** [comment]
Hi @TanishMoral11,

Thank you for your guidance!

I tried installing Bazel using Ubuntu (WSL) on Windows, and I am using the latest version of Android Studio. However, I’m encountering the following error:

![Screenshot 2025-02-15 104548](https://github.com/user-attachments/assets/a7dfe8fe-610a-4d7d-8c18-d0a8159e8aa3)

Steps I Followed:
  Opened Ubuntu (Windows Subsystem for Linux - WSL) terminal.
   Ran the command:
   `sudo apt install bazel-6.5.0
`


Should I use the Bumblebee version of Android Studio, or is the latest version recommended? I also tried using the Bumblebee version, but I’m still facing the same issue.

Any suggestions on how to resolve this? Thanks in advance!

**Reply by TanishMoral11** [reply]
Here's the complete message in raw markdown:

Hi @Varshith-Yadav 
It looks like you're having trouble installing Bazel 6.5.0 because the package is not found in your current repositories. You can follow these steps to install the correct version of Bazel on Ubuntu:

1. **Add the Bazel distribution URI to your package manager:**
   First, you need to add the Bazel distribution URI for Ubuntu to your system.
   ```bash
   sudo apt update && sudo apt install apt-transport-https curl gnupg -y
   curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor >bazel-archive-keyring.gpg
   sudo mv bazel-archive-keyring.gpg /usr/share/keyrings
   sudo apt update
   ```
   
2. **Add the Bazel apt repository:**
   Now, add the Bazel repository for your version of Ubuntu:
   ```bash
   sudo apt install software-properties-common
   sudo apt update
   sudo apt install bazel
   ```
   
3. **Install Bazel 6.5.0:**
   ```bash
   sudo apt install bazel-6.5.0
   ```

4. **Verify Installation:**
   ```bash
   bazel --version
   ```
   
   Also You can use any version of Android Studio, but the recommended version is Bumblebee because the Bazel plugin is stable in that version. If you're still facing issues, it could be helpful to ensure the Bazel plugin is correctly configured in your Android Studio setup.

This should help you resolve the installation issue, but if the error persists, please share more details or screenshots, and I'll be happy to assist you further!

**Reply by Varshith-Yadav** [reply]
Hi @TanishMoral11,

I wanted to let you know that I successfully installed Bazel version 6.5.0. Following the necessary prerequisites for the local setup of this repository, as provided in the repository's wiki, I performed the following steps:

1. **Enabled API levels 28 and 30**
2. **Enabled SDK tools version 32.0.0**
3. **Configured Bazel support**
4. **Set JDK 11 for the project**

Despite following these steps and downloading Bazel, I encountered same error. Could you help me troubleshoot this issue?

**Reply by TanishMoral11** [reply]
Hi @Varshith-Yadav ,  I'm happy to hear that everything is set up and ready for the next step. To proceed, please follow the instructions outlined in the link below:

[Discussion 5672](https://github.com/oppia/oppia-android/discussions/5672)

For running and installing the APK, kindly use the following command:

```bash
$ bazel run //:install_oppia_dev

**Reply by Varshith-Yadav** [reply]
Hello @TanishMoral11,

I am encountering a build error while attempting to compile my Android project. This issue followed two earlier problems I faced with the appt2 and zapalign components. The error appears to be related to unresolved dependencies or version conflicts, and despite attempting multiple fixes, I have been unable to resolve it.

Error Details: 
![Screenshot 2025-02-18 201048](https://github.com/user-attachments/assets/02a451d1-bffb-4c55-b85b-87733cee2544)


Could you please assist me in resolving this build error? If you have any suggestions regarding version compatibility or configuration settings, or if you've faced a similar issue, I would greatly appreciate your help!

**Reply by TanishMoral11** [reply]
Hi @Varshith-Yadav , 
Have you done with these steps in [Discussion 5643](https://github.com/oppia/oppia-android/discussions/5643) for syncing the project with Bazel???

Also make sure that the command is : 

```bash
 bazel run //:install_oppia_dev
